src/main/java/org/jabref/logic/importer/fetcher/ArXivFetcher.java

```diff
@@ -0,0 +1,772 @@
+    private static final String DOI_PREFIX = "10.48550/arXiv.";
+
+    // See https://github.com/JabRef/jabref/pull/9170 discussion
+    /*
+    * Reason behind choice of these fields:
+    *   - KEYWORDS: More descriptive
+    *   - AUTHOR: Better formatted (last name, rest of name)
+    * */
+    private static final Set<Field> CHOSEN_AUTOMATIC_DOI_FIELDS = Set.of(StandardField.KEYWORDS, StandardField.AUTHOR);
+    /*
+     * Reason behind choice of these fields:
+     *   - DOI: give preference to DOIs manually inputted by users, instead of automatic ones
+     *   - PUBLISHER: ArXiv-issued DOIs give 'ArXiv' as entry publisher. While this can be true, prefer using one from external sources,
+     *      if applicable
+     *   - KEY_FIELD: Usually, the KEY_FIELD retrieved from user-assigned DOIs are 'nicer' (instead of a DOI link, it's usually contains one author and the year)
+     * */
+    private static final Set<Field> CHOSEN_MANUAL_DOI_FIELDS = Set.of(StandardField.DOI, StandardField.PUBLISHER, InternalField.KEY_FIELD);
+
+    private static final Map<String, String> ARXIV_KEYWORDS_WITH_COMMA_REPLACEMENTS = ImmutableMap.of(
+            "Computational Engineering, Finance, and Science", "Computational Engineering / Finance / Science",
+            "Distributed, Parallel, and Cluster Computing", "Distributed / Parallel / Cluster Computing");
+
+    private final ArXiv arXiv;
+    private final DoiFetcher doiFetcher;
+
+    public ArXivFetcher(ImportFormatPreferences importFormatPreferences) {
+        this.arXiv = new ArXiv(importFormatPreferences);
+        this.doiFetcher = new DoiFetcher(importFormatPreferences);
+    }
+
+    /**
+     * NOTE: with this constructor, one can disable the additional search for more metadata (with related DOI) by passing
+     * a NULL DoiFetcher.
+     */
+    public ArXivFetcher(ImportFormatPreferences importFormatPreferences, DoiFetcher doiFetcher) {
+        this.arXiv = new ArXiv(importFormatPreferences);
+        this.doiFetcher = doiFetcher;
+    }
+
+    @Override
+    public Optional<URL> findFullText(BibEntry entry) throws IOException {
+        return arXiv.findFullText(entry);
+    }
+
+    @Override
+    public TrustLevel getTrustLevel() {
+        return arXiv.getTrustLevel();
+    }
+
+    @Override
+    public String getName() {
+        return arXiv.getName();
+    }
+
+    @Override
+    public Optional<HelpFile> getHelpPage() {
+        return arXiv.getHelpPage();
+    }
+
+    /**
+     * Remove duplicate values on "KEYWORD" field, if any.
+     *
+     * @param bibEntry A BibEntry to modify
+     */
+    private static void adaptKeywordsFrom(BibEntry bibEntry) {
+        Optional<String> allKeywords = bibEntry.getField(StandardField.KEYWORDS);
+        if (allKeywords.isPresent()) {
+            // See https://github.com/JabRef/jabref/pull/9170/#issuecomment-1266042981
+            for (Map.Entry<String, String> entry : ARXIV_KEYWORDS_WITH_COMMA_REPLACEMENTS.entrySet()) {
```

Comment:
Replacing these cases with a slash my decision on how to circunvent the issue of keywords with commas. Please let me know if you have more fitting solutions.

Reply:
Hi, the issue with library changed by duplicate keywords was fixed. You should use Keywordlist.merge()  You need to get the keyword separator char from the import preferences

Reply:
I'm sorry, but could you expand on what you mean by "use `Keywordlist.merge()`"? Do you mean I should use it for eliminating possible duplicate keywords or for resolving the issue of these comma-separated individual keywords? If its the latter, I don't get how this method could help...

Reply:
Hi, you can use the methods for merging different list of keywords:
 KeywordList.merge(...). This will care about duplicates. 

https://github.com/JabRef/jabref/blob/dcec4e2418530a2d0dbe57360c7636d8eaad9b0d/src/main/java/org/jabref/model/entry/KeywordList.java#L46

Reply:
Ok, done!